src/pages/api/search.ts

```diff
@@ -0,0 +1,49 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { client } from "../../config/elasticsearch";
+import { buildQuery } from "../../helpers/api-functions";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== "POST") {
+    return res.status(400).json({ success: false, message: 'Invalid request method. The endpoint only supports POST requests.' });
```

Comment:
return status code 405
```suggestion
    return res.status(405).json({ success: false, message: 'Invalid request method. The endpoint only supports POST requests.' });
```